version: '3.5'

services:

  mongo:
    image: mongo
    restart: always
    networks:
      - nf-simulator-network
    environment:
      MONGO_INITDB_ROOT_USERNAME: root
      MONGO_INITDB_ROOT_PASSWORD: zXcVbN123!
      MONGO_INITDB_DATABASE: pnf_simulator
    volumes:
      - ./db:/docker-entrypoint-initdb.d
    ports:
      - "27017:27017"

  mongo-express:
    image: mongo-express
    restart: always
    ports:
    - "8081:8081"
    networks:
      - nf-simulator-network
    environment:
      ME_CONFIG_MONGODB_ADMINUSERNAME: root
      ME_CONFIG_MONGODB_ADMINPASSWORD: zXcVbN123!

  ves-client:
    image: onap/org.onap.integration.nfsimulator.vesclient
    ports:
      - "5000:5000"
    networks:
      - nf-simulator-network
    environment:
      USE_CERTIFICATE_FOR_AUTHORIZATION: "false"
    volumes:
      - ./logs:/var/log
      - ./templates:/app/templates
      - ./store:/app/store
      - ./src/main/resources/application.properties:/app/application.properties
    restart: on-failure
    depends_on:
      - mongo
      - mongo-express

networks:
  nf-simulator-network:
    name: nf-simulator-network
